apps/meteor/app/lib/server/functions/processWebhookMessage.ts

```diff
@@ -7,9 +8,39 @@ import { validateRoomMessagePermissionsAsync } from '../../../authorization/serv
 import { getRoomByNameOrIdWithOptionToJoin } from './getRoomByNameOrIdWithOptionToJoin';
 import { sendMessage } from './sendMessage';
 
-export const processWebhookMessage = async function (messageObj, user, defaultValues = { channel: '', alias: '', avatar: '', emoji: '' }) {
+type Payload = {
+	channel?: string | string[];
+	roomId?: string | string[];
+	text?: IMessage['msg'];
+	msg?: IMessage['msg']; // overrided if text is present
+	username?: IMessage['alias'];
+	alias?: IMessage['alias']; // overrided if username is present
+	icon_emoji?: IMessage['emoji'];
+	emoji?: IMessage['emoji']; // overrided if icon_emoji is present
+	icon_url?: IMessage['avatar'];
+	avatar?: IMessage['avatar']; // overrided if icon_url is present
+	attachments?: IMessage['attachments'];
+	parseUrls?: boolean;
+	bot?: IMessage['bot'];
+	groupable?: IMessage['groupable'];
+	tmid?: IMessage['tmid'];
+};
+
+type DefaultValues = {
+	channel: string | string[];
+	alias: string;
+	avatar: string;
+	emoji: string;
+};
+
+export const processWebhookMessage = async function (
+	messageObj: Payload,
+	user: IUser & { username: Exclude<IUser['username'], undefined> },
```

Comment:
We have a utility type for it:

```suggestion
	user: RequiredField<IUser, 'username'>,
```